src/hotspot/share/c1/c1_Canonicalizer.cpp

```diff
@@ -842,14 +842,18 @@ void Canonicalizer::do_LookupSwitch(LookupSwitch* x) {
   if (x->tag()->type()->is_constant()) {
     int v = x->tag()->type()->as_IntConstant()->value();
     BlockBegin* sux = x->default_sux();
-    for (int i = 0; i < x->length(); i++) {
-      int key = x->key_at(i);
-      if (v < key) {
-        break;
-      }
-      if (v == key) {
-        sux = x->sux_at(i);
+    int low = 0;
+    int high = x->length() - 1;
+    while (low <= high) {
+      int mid = low + ((high - low) >> 1);
```

Comment:
Isn't this the same as
`int mid = (low + high) >> 1;`

Reply:
`low + ((high - low) >> 1)`  can avoid integer overflow

Reply:
Good point.  Integer overflow is a real concern.